client/src/components/Rooms.jsx

```
@@ -0,0 +1,150 @@
+'use client';
+
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { MoreVertical } from 'lucide-react';
+
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+
+import './styles.css';
+import { createRoom, deleteRoom, fetchRooms, updateRoom } from '../api/rooms';
+
+export default function Rooms() {
+  const [newRoom, setNewRoom] = useState('');
+  const [editingRoomId, setEditingRoomId] = useState(null);
+  const [editedName, setEditedName] = useState('');
+
+  const navigate = useNavigate();
+  const queryClient = useQueryClient();
+
+  const { data: rooms = [], isLoading } = useQuery({
+    queryKey: ['rooms'],
+    queryFn: fetchRooms,
+  });
```

`fetchRooms` swallows errors and returns `undefined` on failures (client API). Although you default `rooms` to `[]`, consider surfacing query errors to the user (e.g. show an error message) or let the query error so react-query can manage retries. This will make failures more visible instead of silently showing an empty list.